aggregation/performing-complex-mongodb-data-aggregation-queries-with-go.md

```diff
@@ -0,0 +1,231 @@
+# Performing Complex MongoDB Data Aggregation Queries with Go
+
+If you've been following along with my getting started series around MongoDB and Golang, you might remember the tutorial where we took a look at [finding documents](https://www.mongodb.com/blog/post/quick-start-golang--mongodb--how-to-read-documents) in a collection. In this tutorial we saw how to use the `Find` and `FindOne` functions to filter for documents within a collection. This is essentially querying for documents within a specific collection where the filter parameters are fields within the schema of that collection.
+
+So what happens if you need to do something a little more complex like return data that isn't within the schema, do complex manipulations prior to returning a response, or looking across collections in a single command?
+
+This is where the MongoDB aggregation framework becomes valuable.
+
+In this tutorial we're going to look at a few MongoDB aggregation framework examples using the Go programming language, examples that can't really be done with a basic `Find` or `FindOne` operation.
+
+## The Requirements
+
+To be successful with this tutorial, you'll need the following requirements to be met:
+
+- Go 1.10+ installed and configured
+- MongoDB Atlas with an M0 cluster or better
+- The MongoDB Go driver
+
+It is advisable that you've completed the other tutorials in the getting started with MongoDB and Go series as it shares information around the schema that we're using as well as information on connecting to a cluster. However, if you feel comfortable with MongoDB and Go, it isn't an absolute requirement.
```

Comment:
I would add a link to the very first blog post in the quickstart in case a new person stumbles upon this article and wants to start from the beginning.